R/import_metadata_and_counts.R

```diff
@@ -94,10 +94,12 @@ import_metadata_counts <- function(
     select(metadata_tbl, .data$file_id_db) |> mutate(
       original_path = file.path(original_dir, basename(.data$file_id_db)),
       cpm_path = file.path(cache_dir, "cpm", basename(.data$file_id_db))
-    )
+    ) |>
+    distinct()
   
   # Generate cpm from counts
-  cli_alert_info("Generating cpm from {.path {metadata_tbl$file_id_db}}. ")
+  unique_file_ids <- unique(metadata_tbl$file_id_db)
+  cli_alert_info("Generating cpm from {.path {unique_file_ids}}. ")
   get_counts_per_million(input_sce_obj = sce_obj, output_dir = counts_path$cpm_path, hd5_file_dir = counts_path$original_path)
```

Comment:
At this point, `counts_path$cpm_path` seems like it's a vector of filepaths, one for each `file_id_db` in the metadata. However, `get_counts_per_million` expects a single file path. Does this work in practice?

Reply:
Ah yes, I see that `file_id_db = .data$dataset_id |> openssl::md5() |> as.character()`, so it there will only be one `file_id_db`. I think to make this clearer, you could just create `file_id_db` as a separate variable that you add to the metadata data frame, and also use to generate the paths. That way you won't have to filter out the redundant copies at all.

Reply:
Thanks for that, @multimeric; I had a discussion with @myushen  and concluded that an additional argument is not needed at this stage, just good messaging and documentation, in case the user input and multiple data sets in the same single experiment.

Reply:
I wasn't suggesting adding a new argument. I'm just saying that we have a single file ID, which gets *duplicated* as a result of putting it into the data frame, and then we filter it back down again to a single value using `distinct`. This can be avoided by using a separate variable:

```R
file_id_db = .data$dataset_id |> openssl::md5() |> as.character()

...

metadata_tbl <-
    metadata_tbl |> mutate(file_id_db = file_id_db)

...


original_path <- file.path(original_dir, basename(file_id_db))
cpm_path <- file.path(cache_dir, "cpm", basename(file_id_db))
```

Reply:
but the redundant file_if_db column will be carried in the metadata, that is what we use to do future queries. So, in the metadata, the redundancy will remain. 

Or am I missing something?

Reply:
Yes, I realise it will be like that in the metadata. My suggestion is just to make the code simpler and clearer.

Reply:
sure in the backend @myushen follow @multimeric lead.

I think @myushen intended that as modifying the front end (code interface to the user)

Reply:
I will add a check point at the front end and rename the function for clarification